noaa_metrics/aggregate_logs.py

```diff
@@ -0,0 +1,208 @@
+    by_location_df = by_location_df_raw.rename(
+        columns={
+            "nunique": "Distinct Users",
+            "count": "Files Sent",
+            "sum": "Download Volume (MB)",
+        }
+    )
+    by_location_df.index = by_location_df.index.rename("Domain Type")
+    by_location_df.loc["Total"] = by_location_df.sum()
+    return by_location_df
+
+
+def df_to_csv(df: pd.DataFrame, header: str, output_csv):
+    with open(output_csv, "a") as file:
+        file.write(header)
+        df.to_csv(file, header=True, index=True)
+
+
+def get_month(date):
+    month = calendar.month_name[(date.month)]
+    return month
+
+
+def get_year(date):
+    year = date.year
+    return year
+
+
+def email_full_report(full_report, year, start_month, end_month, mailto: str, dataset):
+    if start_month == end_month and dataset == "all":
+        subject = f"NOAA Downloads {start_month} {year}"
+        filename = f"NOAA-{start_month}-{year}.csv"
+    elif start_month != end_month and dataset == "all":
+        subject = f"NOAA Downloads {start_month} - {end_month} {year}"
+        filename = f"NOAA-{start_month}-{end_month}-{year}.csv"
+    elif start_month == end_month and dataset != "all":
+        subject = f"NOAA Downloads {dataset} {start_month} - {end_month} {year}"
+        filename = f"NOAA-{dataset}-{start_month}-{end_month}-{year}.csv"
+    elif start_month != end_month and dataset != "all":
+        subject = f"NOAA Downloads {dataset} {start_month} {year}"
+        filename = f"NOAA-{dataset}-{start_month}-{year}.csv"
+
+    msg = EmailMessage()
+    msg["From"] = "archive@nusnow.colorado.edu"
+    msg["To"] = mailto
+    msg["Subject"] = subject
+
+    with open(full_report) as fp:
+        metrics_data = fp.read()
+    msg.add_attachment(metrics_data, filename=filename)
+    with smtplib.SMTP("localhost") as s:
+        s.send_message(msg)
+
+
+def aggregate_logs(start_date, end_date, mailto: str, dataset:str):
+    start_date_str = start_date.isoformat()
+    end_date_str = end_date.isoformat()
+    log_df = create_dataframe(JSON_OUTPUT_DIR, start_date, end_date)
+
+    if dataset != "all":
+        log_df = filter_to_dataset(log_df, dataset)
+
+    start_month = get_month(start_date)
+    end_month = get_month(end_date)
+    year = get_year(start_date)
+    summary_df = get_period_summary_stats(log_df)
+    by_dataset_df = downloads_by_dataset(log_df)
+    by_day_df = downloads_by_day(log_df)
+    by_location_df = downloads_by_tld(log_df)
+
+    if start_month == end_month:
+        if dataset != "all":
+            summary_header = f"NOAA Downloads {dataset} {start_month}\n\n"
+        else:
+            summary_header = f"NOAA Downloads {start_month}\n\n"
+    else:
+        if dataset != "all":
+            summary_header = f"NOAA Downloads {dataset} {start_month} - {end_month}\n\n"
+        else:
+            summary_header = f"NOAA Downloads {start_month} - {end_month}\n\n"
```

Comment:
If possible, I'd like to combine this conditional tree with the one above. The ifs and elses are all the same, so maybe we could be setting the `subject`, `filename`, and `summary_header` all in one place.

Reply:
agreed! Done!